tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -381,3 +383,75 @@ def get_tag_value(self, i):
         if tag_value is not False:
             tag_value = tag_value.lower().strip("'\"")
         return tag_value
+
+
+class FirewallRulesFilter(Filter):
+
+    schema = type_schema(
+        'firewall-rules',
+        **{
+            'include': {'type': 'array', 'items': {'type': 'string'}},
+            'equal': {'type': 'array', 'items': {'type': 'string'}}
+        })
+
+    def __init__(self, data, manager=None):
+        super(FirewallRulesFilter, self).__init__(data, manager)
+        self.policy_include = None
+        self.policy_equal = None
+        self.log = logging.getLogger('custodian.azure')
+
+    def validate(self):
+        self.policy_include = IpRangeHelper.parse_ip_ranges(self.data, 'include')
+        self.policy_equal = IpRangeHelper.parse_ip_ranges(self.data, 'equal')
+
+        has_include = self.policy_include is not None
+        has_equal = self.policy_equal is not None
+
+        if has_include and has_equal:
+            raise FilterValidationError('Cannot have both include and equal.')
+
+        if not has_include and not has_equal:
+            raise FilterValidationError('Must have either include or equal.')
+
+        return True
+
+    def process(self, resources, event=None):
+        result, _ = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._check_resources,
+            executor_factory=self.executor_factory,
+            log=self.log
+        )
+
+        return result
+
+    def _check_resources(self, resources, event):
+        return [r for r in resources if self._check_resource(r)]
+
+    def _query_rules(self, resource):
+        """
+        Queries firewall rules for a resource. Override in concrete classes.
+        :param resource:
+        :return: A set of netaddr.IPRange or netaddr.IPSet with rules defined for the resource.
+        """
+        return []
+
+    def _check_resource(self, resource):
+        try:
+            resource_rules = self._query_rules(resource)
+        except Exception as error:
+            self.log.warning(error)
```

Comment:
I prefer raising exception instead of logging an error. 
Second option is to raise it by default but allow child classes to override the behavior with 'log and move forward'

Reply:
ok